src/include/btree.i

```diff
@@ -496,19 +498,21 @@ __wt_page_only_modify_set(WT_SESSION_IMPL *session, WT_PAGE *page)
 	WT_ASSERT(session, !F_ISSET(session->dhandle, WT_DHANDLE_DEAD));
 
 	last_running = 0;
-	if (page->modify->write_gen == 0)
+	if (page->modify->page_state == WT_PAGE_CLEAN)
 		last_running = S2C(session)->txn_global.last_running;
 
 	/*
-	 * We depend on atomic-add being a write barrier, that is, a barrier to
+	 * We depend on atomic-cas being a write barrier, that is, a barrier to
```

Comment:
This should go back to being `atomic-add` or I'd prefer `the atomic operation` - then we won't need to keep changing it.

I'd like a comment in here about wrapping. Something like:

```
The page state can only ever be incremented above dirty by the number of concurrently running threads, so the counter will never approach the point where it would wrap.
```

Reply:
4c3cd34